go/vt/vtgate/planbuilder/operators/plan_query.go

```diff
@@ -73,13 +74,28 @@ func PlanQuery(ctx *plancontext.PlanningContext, stmt sqlparser.Statement) (resu
 	checkValid(op)
 	op = planQuery(ctx, op)
 
-	_, isRoute := op.(*Route)
-	if !isRoute && ctx.SemTable.NotSingleRouteErr != nil {
-		// If we got here, we don't have a single shard plan
-		return nil, ctx.SemTable.NotSingleRouteErr
+	err = shouldErrorOnComplexPlan(ctx, op)
+	return op, err
+}
+
+func shouldErrorOnComplexPlan(ctx *plancontext.PlanningContext, op Operator) error {
```

Comment:
`shouldErrorOnComplexPlan` is a bit confusing - we don't fail on complex plans, we just pass through simple plans :)

how about `checkNotSingleRouteErr` or something along those lines?